Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -790,43 +790,60 @@ namespace OpenInfraPlatform {
 					}
 				}
 				
+				/*! \brief  Converts a list of \c IfcFaceBound -s to a list of boundary loops.
+				 * The first loop in the return list is converted from the \c IfcFaceOuterBound entity.
+				 * Only one boundary should be of this type.
+				 *
+				 * \param[in]	ifcFaceBounds	List of \c IfcFaceBound entities to be converted.
+				 * \param[in]	pos				The relative location of the origin of the representation's coordinate system within the geometric context.
+				 * \return		List of boundary loops.
+				 */
 				std::vector<std::vector<carve::geom::vector<3>>> convertIfcFaceBoundList(
-					const std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>>& ifcFaceBounds,
+					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>> ifcFaceBounds,

```

Comment:
I'd leave the reference `&` - this way the vector is not unnecessarily copied...

Reply:
With the reference `&`, the variable `ifcFaceBounds` becoms a const vector, because the source `face->Bounds` in `convertIfcFace` (line 775) is a const variable. (`const EXPRESSReference<typename IfcEntityTypesT::IfcFace>& face`, line 763)
At least, the function `swapOuterBoundaryToFront` (called in line 805) needs a non-const vector in order to perform the swap of elements. Probably, we need the copy at some point to come from const to non-const? (see message of commit cc2b38a)
By just adding the `&` in line 802 (where you commented), the compiler complains "cannot convert argument 1 from 'const (...)' to 'std::vector<(...)> &' "

Reply:
Right, then it needs copying, I see. I withdraw my comment.